### ISSUE TITLE
Error when using auth: Cannot read properties of undefined

### ISSUE THREAD

**Issue by crowbait** [cdimascio/express-openapi-validator#1005]
**Describe the bug**
When using BasicAuth in my spec, `express-openapi-validator` fails with the following message:
```
Cannot read properties of undefined (reading 'undefined')
```

Validator settings as follows:
```ts
app.use(OpenApiValidator.middleware({
  apiSpec: specYAML /* const, path to spec generated on runtime */,
  validateRequests: {
    coerceTypes: false,
    allowUnknownQueryParameters: false,
    removeAdditional: 'all'
  },
  validateResponses: {
    coerceTypes: false,
    removeAdditional: 'failing'
  }
}));
```

I'm using the following spec (cut for consistency):
```yaml
openapi: 3.0.0
paths:
  /auth:
    get:
      operationId: auth
      description: Accepts credentials in HTTP Basic Auth and validates them. Will respond with HTTP status codes but no content.
      parameters: []
      responses:
        '200':
          description: The request has succeeded.
        '401':
          description: Access is unauthorized.
      security:
        - BasicAuth: []
  securitySchemes:
    BasicAuth:
      type: http
      scheme: basic
```

Using the same security scheme on any other route fails with the same message. Removing the `security` description from the spec "solves" the issue.

**Comment by cdimascio** [comment]
This sounds similar to
https://github.com/cdimascio/express-openapi-validator/pull/1003 which is
fixed in 5.3.9.

**Comment by crowbait** [comment]
> This sounds similar to #1003 which is fixed in 5.3.9.

You are *absolutely* right.

I severy misinterpreted that issue - I thought "this is about cookies which I don't use, this can't possibly affect me" ... without realizing the issue was actually about a *lack* of cookies on an auth-enabled request. It is indeed fixed in 5.3.9 .

Sorry for not thinking this one through.

**Comment by cdimascio** [comment]
No glad its resolved. Cheers
